### PR TITLE
find system Python installations on Windows

### PR DESCRIPTION
**Pull request by kevinushey**
### Intent

Addresses https://github.com/rstudio/rstudio/issues/9379.

### Approach

We had a routine intended to find "system" installations of Python, which lacked the Windows-specific pieces necessary. This PR adds those bits.

### Automated Tests

None included (no infrastructure for Windows tests yet).

### QA Notes

Test via notes in https://github.com/rstudio/rstudio/issues/9379.

### Checklist

- [x] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [x] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [x] This PR passes all local unit tests

### CODE REVIEW

File: src/cpp/session/modules/SessionPythonEnvironments.R [hunk 1]
```diff
@@ -130,6 +130,9 @@
    # prefer UTF-8 path when possible
    path <- enc2utf8(path)
    
+   # prefer unix-style slashes
+   path <- chartr("\\", "/", path)
```
Comment (jmcphers) [inline src/cpp/session/modules/SessionPythonEnvironments.R:134]: TIL `chartr`!

**Review by jmcphers — APPROVED** [review]
Does this need a NEWS entry?

**Comment by jmcphers** [comment]
Was this change intended for the Juliet Rose branch?

**Comment by kevinushey** [comment]
It was -- thanks for catching that. If only GitHub were smart enough to infer that I wanted to merge this branch into its "parent" branch...

**Pull request merged**